components/odh-notebook-controller-webhook/controllers/notebook_webhook.go

```diff
@@ -0,0 +1,399 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	configv1 "github.com/openshift/api/config/v1"
+	"net/http"
+
+	nbv1 "github.com/kubeflow/kubeflow/components/notebook-controller/api/v1"
+	"github.com/kubeflow/kubeflow/components/notebook-controller/pkg/culler"
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+//+kubebuilder:webhook:path=/mutate-notebook-v1,mutating=true,failurePolicy=fail,sideEffects=None,groups=kubeflow.org,resources=notebooks,verbs=create;update,versions=v1,name=notebooks.opendatahub.io,admissionReviewVersions=v1
+
+// NotebookWebhook holds the webhook configuration.
+type NotebookWebhook struct {
+	Client      client.Client
+	decoder     *admission.Decoder
+	OAuthConfig OAuthConfig
+}
+
+var proxyEnvVars = make(map[string]string, 4)
+
+// InjectReconciliationLock injects the kubefllow notebook controller culling
+// stop annotation to explicitly start the notebook pod when the ODH notebook
+// controller finishes the reconciliation. Otherwise a race condition may happen
+// while mounting the notebook service account pull secret into the pod.
+//
+// The ODH notebook controller will remove this annotation when the first
+// reconcilitation is completed (see RemoveReconciliationLock).
+func InjectReconciliationLock(meta *metav1.ObjectMeta) error {
+	if meta.Annotations != nil {
+		meta.Annotations[culler.STOP_ANNOTATION] = AnnotationValueReconciliationLock
```

Comment:
This should be `controllers.AnnotationValueReconciliationLock`. There are couple of more instances below as well.